alkira/resource_alkira_service_fortinet.go

```diff
@@ -217,14 +217,20 @@ func resourceFortinetDelete(d *schema.ResourceData, m interface{}) error {
 	return client.DeleteFortinet(d.Id())
 }
 
-func generateFortinetRequest(d *schema.ResourceData, m interface{}) (*alkira.Fortinet, error) {
+func generateFortinetRequest(d *schema.ResourceData, gs getSegmentById) (*alkira.Fortinet, error) {
 	billingTagIds := convertTypeListToIntList(d.Get("billing_tag_ids").([]interface{}))
 	managementServer := &alkira.FortinetManagmentServer{
 		IpAddress: d.Get("management_server_ip").(string),
 		Segment:   d.Get("management_server_segment").(string),
 	}
 	instances := expandFortinetInstances(d.Get("instances").(*schema.Set))
-	segmentNames := convertTypeListToStringList(d.Get("segment_names").([]interface{}))
+
+	// convert segment ids to segment names
```

Comment:
This could be one global helper function, nope? We always take segment_ids and convert to segment_names